### PR TITLE
fix(web-app): set unique id for treeview nodes

### LINKED ISSUE #1285: Clicking on "About Carbon" anchor link or directly on the heading is opening the sidenav
**Bug 🐛** Clicking on "About Carbon" anchor link or directly on the heading is opening the sidenav (see attached video). We think it's due to targeting the "About Carbon" inside of the sidenav.
**Page:**: https://next.carbondesignsystem.com/about-carbon/help/faq
**Browser:** All (tested on Chrome, Firefox & Safari)
**Suggestion:** Should not open the sidenav and scroll to paragraph instead.

https://user-images.githubusercontent.com/40550942/190458875-021f41b8-e59b-48c5-83d5-ddeae12b4ddd.mov

_Originally posted by @francinelucca in https://github.com/carbon-design-system/carbon-platform/discussions/1268#discussioncomment-3655745_
Screen Shot 2022-09-20 at 8 19 33 AM

### PR DESCRIPTION
**Pull request by alisonjoseph**
closes #1285

#### Testing / reviewing

Ensure nav works correctly, and that nodes have unique ids so anchor link works correctly on FAQ / About Carbon Page and doesn't interact with left nav.

http://localhost:3000/about-carbon/help/faq

@jdharvey-ibm I went back and re-added your changes to this file

### CODE REVIEW

**Review by jdharvey-ibm — APPROVED** [review]

**Review by francinelucca — CHANGES_REQUESTED** [review]
This is messing with the activeItem now 🤔 , not sure how to fix but let's pair on this if you want!
image

**Review by francinelucca — APPROVED** [review]
💃

File: services/web-app/components/nav-tree/nav-tree.js [hunk 1]
```diff
@@ -35,6 +35,18 @@ const NavTree = ({ activeItem, items = [], label }) => {
...
+    dfs(newItemNodeArray, (evalNode) => {
+      if (!evalNode.parentNodeId) evalNode.parentNodeId = 'left_nav_tree'
+      evalNode.id = `${evalNode.parentNodeId}_${evalNode.title
+        .toLocaleLowerCase()
+        .replace(/\s/g, '')}`
+      evalNode.items?.forEach((child) => {
+        child.parentNodeId = evalNode.id
+      })
+      return false
+    })
```
Comment (jdharvey-ibm) [inline services/web-app/components/nav-tree/nav-tree.js:49]: Suggestion to help justify this code to "future us".

```suggestion
    // Create hierarchical, unique node ids for all nodes in the nav tree
    dfs(newItemNodeArray, (evalNode) => {
      if (!evalNode.parentNodeId) {
        evalNode.parentNodeId = 'left_nav_tree'
      }
      
      // Combine the parent's node id with the current node's id
      const currentNodeId = evalNode.title.toLocaleLowerCase().replace(/\s/g, '')
      evalNode.id = `${evalNode.parentNodeId}_${currentNodeId}`
      
      // Set the parent node id of each child of this node to the newly generated id
      evalNode.items?.forEach((child) => {
        child.parentNodeId = evalNode.id
      })
      
      return false
    })
```

Comment (francinelucca) [reply services/web-app/components/nav-tree/nav-tree.js:49]: thanks!

**Pull request merged**